generator/beat/{beat}/Makefile

```diff
@@ -2,47 +2,84 @@ BEAT_NAME={beat}
 BEAT_PATH={beat_path}
 BEAT_GOPATH=$(firstword $(subst :, ,${GOPATH}))
 BEAT_URL=https://${BEAT_PATH}
-SYSTEM_TESTS=false
-TEST_ENVIRONMENT=false
-ES_BEATS?=./vendor/github.com/elastic/beats
-GOPACKAGES=$(shell govendor list -no-status +local)
-PREFIX?=.
-NOTICE_FILE=NOTICE
-
-# Path to the libbeat Makefile
--include $(ES_BEATS)/libbeat/scripts/Makefile
+BEAT_DOC_URL=https://path/to/{beat}/doc
+BEAT_DESCRIPTION=My description
+BEAT_LICENSE=My license
+COPYRIGHT_TXT=My company
+COPYRIGHT_YEAR={current_year}
+
+ES_BEATS=./_beats
+PREFIX=.
+BEATS_VERSION?=master # For specific version: BEATS_VERSION?=6.1
+NOTICE_FILE=NOTICE.txt
+LICENSE_FILE=LICENSE.txt
+NOW=$(shell date -u '+%Y-%m-%dT%H:%M:%S')
+GOBUILD_FLAGS=-i -ldflags "-s -X $(BEAT_PATH)/version.buildTime=$(NOW) -X $(BEAT_PATH)/version.commit=$(COMMIT_ID)"
 
-# Initial beat setup
-.PHONY: setup
-setup: copy-vendor
-	$(MAKE) update
 
-# Copy beats into vendor directory
-.PHONY: copy-vendor
-copy-vendor:
-	mkdir -p vendor/github.com/elastic/
-	cp -R ${BEAT_GOPATH}/src/github.com/elastic/beats vendor/github.com/elastic/
-	rm -rf vendor/github.com/elastic/beats/.git
+SYSTEM_TESTS=true
+TEST_ENVIRONMENT=true
+TESTING_ENVIRONMENT=snapshot-noxpack
 
-.PHONY: git-init
-git-init:
+# Path to the libbeachmt Makefile
+-include $(ES_BEATS)/libbeat/scripts/Makefile
+
+setup:
+	$(MAKE) update-dependencies
 	git init
-	git add README.md CONTRIBUTING.md
-	git commit -m "Initial commit"
-	git add LICENSE
-	git commit -m "Add the LICENSE"
-	git add .gitignore
-	git commit -m "Add git settings"
-	git add .
-	git reset -- .travis.yml
-	git commit -m "Add {beat}"
-	git add .travis.yml
-	git commit -m "Add Travis CI"
+
+# updates {beat} dependencies in the vendor folder
+vendor:
+	mkdir -p vendor
+	govendor init
+
+update-dependencies: vendor
+	$(MAKE) update-libbeat
+	$(MAKE) update-{beat}-dependencies
+
+update-libbeat:
+	@chmod +x scripts/clean_vendor.sh
```

Comment:
It's pretty hard to review in which case it's going to work and when not. Best lets get this in an play with it :-)